tests/api/crud/test_secrets.py

```diff
@@ -31,6 +35,415 @@ def test_store_secrets_verifications(
+                provider=provider, secrets={"invalid/key": "value"}
+            ),
+            key_map_secret_key="invalid-key-map-secret-key",
+        )
+
+    # internal not allowed
+    with pytest.raises(mlrun.errors.MLRunAccessDeniedError):
+        mlrun.api.crud.Secrets().store_secrets(
+            project,
+            mlrun.api.schemas.SecretsData(
+                provider=provider, secrets={"invalid/key": "value"}
+            ),
+            key_map_secret_key=key_map_secret_key,
+        )
+
+
+def test_get_secret_verifications(
+    db: sqlalchemy.orm.Session,
+    client: fastapi.testclient.TestClient,
+    k8s_secrets_mock: tests.api.conftest.K8sSecretsMock,
+):
+    project = "project-name"
+    provider = mlrun.api.schemas.SecretProviderName.kubernetes
+    key_map_secret_key = mlrun.api.crud.Secrets().generate_schedule_key_map_secret_key()
+
+    # verifications check
+    # not allowed from k8s
+    with pytest.raises(mlrun.errors.MLRunAccessDeniedError):
+        mlrun.api.crud.Secrets().get_secret(project, provider, "does-not-exist-key")
+
+    # key map with provider other than k8s
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+        mlrun.api.crud.Secrets().get_secret(
+            project,
+            mlrun.api.schemas.SecretProviderName.vault,
+            "does-not-exist-key",
+            key_map_secret_key=key_map_secret_key,
+        )
+
+
+def test_get_secret(
+    db: sqlalchemy.orm.Session,
+    client: fastapi.testclient.TestClient,
+    k8s_secrets_mock: tests.api.conftest.K8sSecretsMock,
+):
+    _mock_secrets_crud_uuid_generation()
+    project = "project-name"
+    provider = mlrun.api.schemas.SecretProviderName.kubernetes
+    key_map_secret_key = mlrun.api.crud.Secrets().generate_schedule_key_map_secret_key()
+    invalid_secret_key = "invalid/key"
+    invalid_secret_value = "some-value"
+    invalid_secret_2_key = "invalid/key/2"
+    invalid_secret_2_value = "some-value-3"
+    valid_secret_key = "valid-key"
+    valid_secret_value = "some-value-5"
+
+    # sanity - none returned on keys that does not exist
+    assert (
+        mlrun.api.crud.Secrets().get_secret(
+            project, provider, "does-not-exist-key", allow_secrets_from_k8s=True
+        )
+        is None
+    )
+    assert (
+        mlrun.api.crud.Secrets().get_secret(
+            project,
+            provider,
+            "does-not-exist-key",
+            allow_secrets_from_k8s=True,
+            allow_internal_secrets=True,
+            key_map_secret_key=key_map_secret_key,
+        )
+        is None
+    )
+
+    mlrun.api.crud.Secrets().store_secrets(
+        project,
+        mlrun.api.schemas.SecretsData(
+            provider=provider,
+            secrets={
+                valid_secret_key: valid_secret_value,
+                invalid_secret_key: invalid_secret_value,
+                invalid_secret_2_key: invalid_secret_2_value,
+            },
+        ),
+        True,
```

Comment:
```suggestion
        allow_secrets_from_k8s=True,
```
Easier to follow if you specify named argument here.